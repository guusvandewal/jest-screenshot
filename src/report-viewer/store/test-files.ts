import { component } from "tsdi";
import { observable, action, computed } from "mobx";
import { FileReport } from "../../reporter-types";

@component
export class StoreTestFiles {
    @observable public activeFileName = testResults.files.length > 0 ? testResults.files[0].testFilePath : undefined;

    @action.bound public selectFile(file: FileReport) {
        this.activeFileName = file.testFilePath;
    }

    @computed public get activeFile() {
        return testResults.files.find(file => file.testFilePath === this.activeFileName);
    }

    public isActive(file: FileReport) {
        return this.activeFileName === file.testFilePath;
    }
}
