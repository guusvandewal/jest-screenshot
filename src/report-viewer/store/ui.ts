import { component } from "tsdi";
import { observable, action, computed } from "mobx";
import { FileReport } from "../../reporter-types";
import { bind } from "lodash-decorators";

@component
export class StoreUi {
    @observable public menuVisible = true;

    @bind @action public toggleMenu() {
        this.menuVisible = !this.menuVisible;
    }
}
