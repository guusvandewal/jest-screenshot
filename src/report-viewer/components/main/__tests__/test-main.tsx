import * as React from "react";
import { shallow } from "enzyme";
import { Main } from "..";
import { StoreUi } from "../../../store";

describe("Main", () => {
    it("looks as expected", () => {
        expect(shallow(<Main />)).toMatchSnapshot();
    });

    it("looks as expected with no reports", () => {
        testResults.files = [];
        expect(shallow(<Main />)).toMatchSnapshot();
    });

    it("toggles the menu visibility", () => {
        const element = shallow(<Main />);
        expect(element.find("main").hasClass("is-9")).toBe(true);
        expect(element.find("main").hasClass("is-12")).toBe(false);

        tsdi.get(StoreUi).toggleMenu();
        element.update();
        expect(element).toMatchSnapshot();
        expect(element.find("main").hasClass("is-9")).toBe(false);
        expect(element.find("main").hasClass("is-12")).toBe(true);

        tsdi.get(StoreUi).toggleMenu();
        element.update();
        expect(element).toMatchSnapshot();
        expect(element.find("main").hasClass("is-9")).toBe(true);
        expect(element.find("main").hasClass("is-12")).toBe(false);
    });
});
