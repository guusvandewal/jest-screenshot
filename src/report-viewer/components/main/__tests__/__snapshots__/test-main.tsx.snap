// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main looks as expected 1`] = `
<main
  className="column is-9 section is-fullheight"
>
  <h1
    className="title"
  >
    src/__tests__/test-first.ts
  </h1>
  <FailedTestSection
    failedTest={
      Object {
        "failedSnapshots": Array [
          Object {
            "changedPixels": 5,
            "changedRelative": 0.5,
            "diffPath": "jest-screenshot-report/reports/some-identifier.snap.png/diff.png",
            "height": 2,
            "message": "Received message.",
            "receivedPath": "jest-screenshot-report/reports/some-identifier.snap.png/received.png",
            "snapshotNumber": 1,
            "snapshotPath": "jest-screenshot-report/reports/some-identifier.snap.png/snapshot.png",
            "testFileName": "src/__tests__/test-first.ts",
            "testName": "The first snapshots",
            "totalPixels": 10,
            "width": 5,
          },
          Object {
            "changedPixels": 20,
            "changedRelative": 0.2,
            "diffPath": "jest-screenshot-report/reports/some-other-identifier.snap.png/diff.png",
            "height": 10,
            "message": "Received message.",
            "receivedPath": "jest-screenshot-report/reports/some-other-identifier.snap.png/received.png",
            "snapshotNumber": 2,
            "snapshotPath": "jest-screenshot-report/reports/some-other-identifier.snap.png/snapshot.png",
            "testFileName": "src/__tests__/test-first.ts",
            "testName": "The first snapshots",
            "totalPixels": 100,
            "width": 10,
          },
        ],
        "titles": Array [
          "The thing",
          "with this other thing",
          "does things",
        ],
      }
    }
    key="The thing-with this other thing-does things"
  />
</main>
`;

exports[`Main looks as expected with no reports 1`] = `""`;

exports[`Main toggles the menu visibility 1`] = `
<main
  className="column is-12 section is-fullheight"
>
  <h1
    className="title"
  >
    src/__tests__/test-first.ts
  </h1>
  <FailedTestSection
    failedTest={
      Object {
        "failedSnapshots": Array [
          Object {
            "changedPixels": 5,
            "changedRelative": 0.5,
            "diffPath": "jest-screenshot-report/reports/some-identifier.snap.png/diff.png",
            "height": 2,
            "message": "Received message.",
            "receivedPath": "jest-screenshot-report/reports/some-identifier.snap.png/received.png",
            "snapshotNumber": 1,
            "snapshotPath": "jest-screenshot-report/reports/some-identifier.snap.png/snapshot.png",
            "testFileName": "src/__tests__/test-first.ts",
            "testName": "The first snapshots",
            "totalPixels": 10,
            "width": 5,
          },
          Object {
            "changedPixels": 20,
            "changedRelative": 0.2,
            "diffPath": "jest-screenshot-report/reports/some-other-identifier.snap.png/diff.png",
            "height": 10,
            "message": "Received message.",
            "receivedPath": "jest-screenshot-report/reports/some-other-identifier.snap.png/received.png",
            "snapshotNumber": 2,
            "snapshotPath": "jest-screenshot-report/reports/some-other-identifier.snap.png/snapshot.png",
            "testFileName": "src/__tests__/test-first.ts",
            "testName": "The first snapshots",
            "totalPixels": 100,
            "width": 10,
          },
        ],
        "titles": Array [
          "The thing",
          "with this other thing",
          "does things",
        ],
      }
    }
    key="The thing-with this other thing-does things"
  />
</main>
`;

exports[`Main toggles the menu visibility 2`] = `
<main
  className="column is-9 section is-fullheight"
>
  <h1
    className="title"
  >
    src/__tests__/test-first.ts
  </h1>
  <FailedTestSection
    failedTest={
      Object {
        "failedSnapshots": Array [
          Object {
            "changedPixels": 5,
            "changedRelative": 0.5,
            "diffPath": "jest-screenshot-report/reports/some-identifier.snap.png/diff.png",
            "height": 2,
            "message": "Received message.",
            "receivedPath": "jest-screenshot-report/reports/some-identifier.snap.png/received.png",
            "snapshotNumber": 1,
            "snapshotPath": "jest-screenshot-report/reports/some-identifier.snap.png/snapshot.png",
            "testFileName": "src/__tests__/test-first.ts",
            "testName": "The first snapshots",
            "totalPixels": 10,
            "width": 5,
          },
          Object {
            "changedPixels": 20,
            "changedRelative": 0.2,
            "diffPath": "jest-screenshot-report/reports/some-other-identifier.snap.png/diff.png",
            "height": 10,
            "message": "Received message.",
            "receivedPath": "jest-screenshot-report/reports/some-other-identifier.snap.png/received.png",
            "snapshotNumber": 2,
            "snapshotPath": "jest-screenshot-report/reports/some-other-identifier.snap.png/snapshot.png",
            "testFileName": "src/__tests__/test-first.ts",
            "testName": "The first snapshots",
            "totalPixels": 100,
            "width": 10,
          },
        ],
        "titles": Array [
          "The thing",
          "with this other thing",
          "does things",
        ],
      }
    }
    key="The thing-with this other thing-does things"
  />
</main>
`;
